observation_portal/common/state_changes.py

```diff
@@ -225,15 +230,21 @@ def modify_ipp_time_from_requests(ipp_val, requests_list, modification='debit'):
         logger.warning(_(f'Problem {modification}ing ipp time for request {request.id}: {repr(e)}'))
 
 
-def get_request_state_from_configuration_statuses(request_state, acceptability_threshold, configuration_statuses):
+def get_request_state_from_configuration_statuses(old_request_state, request, configuration_statuses):
     """Determine request state from all the configuration statuses associated with one of the request's observations"""
+    acceptability_threshold = request.acceptability_threshold
     observation_state = get_observation_state(configuration_statuses)
     completion_percent = exposure_completion_percentage(configuration_statuses)
     if isclose(acceptability_threshold, completion_percent) or \
             completion_percent >= acceptability_threshold or \
             observation_state == 'COMPLETED':
         return 'COMPLETED'
-    return request_state
+    # If a nonzero MAX_RETRIES_PER_REQUEST is set and the observation failed, check that condition
+    if settings.MAX_RETRIES_PER_REQUEST and observation_state == 'FAILED':
```

Comment:
I'm only counting FAILED observations for the retry limit, because I think FAILED is the only state that occurs due to the observation guiding/acquisition type issues we are trying to avoid here. ABORTED should only occur when it's forcefully cancelled by the scheduler for another observation, so that shouldn't count towards the limit.

Reply:
Sounds good to me